### PR TITLE
Add beforeSubmit hook

### PR DESCRIPTION
**Pull request by offirgolan**
There are many times where we may want to normalize data before we validate/submit

### CODE REVIEW

**Comment by offirgolan** [comment]
@simonihmig, I made another change to pass available arguments to their actions. This is necessary if you're dependent on a server returned error message or some specific post validation check.

**Comment by simonihmig** [comment]
Thanks again @offirgolan! The hook and passing the results of validations to the actions does make sense. Why is passing the raw event object `e` necessary? It is a rather low level thing I would tend to not pass around to the app unless absolutely necessary for some use cases...

Things that should be added before merging:

- [ ] mention the beforeSubmit action and the passing of the results to docs ("Submitting the form")
- [ ] add a simple test for asserting that beforeSubmit is called

**Comment by offirgolan** [comment]
@simonihmig I wasn't sure it was really necessary or not tbh but ill remove it. To follow your naming convention, do you want to call this action `before` or keep it as `beforeSubmit`?

**Comment by offirgolan** [comment]
@simonihmig Added docs and tests. Also removed `e` from action calls and renamed `beforeSubmit` to `before`.

File: addon/components/bs-form.js [hunk 1]
```diff
@@ -161,6 +175,7 @@ export default Ember.Component.extend({
    * messages will be shown automatically.
    *
    * @event submit
+   * @param Object result The returned result from the validate method
```
Comment (simonihmig) [inline addon/components/bs-form.js]: This added @param statement is incorrect. `e` ist the DOM submit event.

**Comment by offirgolan** [comment]
@simonihmig fixed it by exposing `action` as a public action handler.

**Comment by simonihmig** [comment]
Travic build was failing because of some trivial JSCS code styling issues, please fix them if still present....

**Comment by offirgolan** [comment]
Should be fixed now

**Pull request merged**